### PR TITLE
Fixed Lua wrapper warnings about signed/unsigned comparisons

### PR DESCRIPTION
**Pull request by dgant**
https://github.com/TorchCraft/TorchCraft/issues/218

Compiling TorchCraft produced a bunch of warnings about signed/unsigned integer comparison. Now it doesn't.

@jgehring @ebetica @syhw

### CODE REVIEW

File: lua/state_lua.cpp [hunk 1]
```diff
@@ -256,7 +256,7 @@ void updateUserValueAndPush(
 
   // Still up to date?
   lua_getfield(L, -1, "__numUpdates");
-  if (lua_tointeger(L, -1) == s->numUpdates) {
+  if ((long) lua_tointeger(L, -1) == (long) s->numUpdates) {
```
Comment (ebetica) [inline lua/state_lua.cpp]: Maybe lua_tolong here

Comment (dgant) [reply lua/state_lua.cpp]: lua_tolong seems to be unavailable. Will use lua_tonumber https://github.com/quakenet/newserv/blob/master/lua/lua.h#L91

Comment (ebetica) [reply lua/state_lua.cpp]: Ah I didn't realize. tonumber appears to give a double, so they original might be better. Sorry about that!

Comment (dgant) [reply lua/state_lua.cpp]: #define lua_toint(a, b) (int) lua_tonumber(a, b)

so (long) lua_tonumber(a, b) is still better than what I originally had :)

Comment (ebetica) [reply lua/state_lua.cpp]: Ahaha wtf, feel free to merge then :)

Comment (dgant) [reply lua/state_lua.cpp]: BTW if you want a totally sweet probably-not-functional Lua wrapper for BWAPI 2.3 -- I wrote https://code.google.com/archive/p/bwapi-lua/ back in like 2009 :D (and proceeded to never use it)

**Pull request merged**